ts/Stock/Navigator/StandaloneNavigator.ts

```diff
@@ -0,0 +1,330 @@
+/* *
+ *
+ *  (c) 2010-2023 Mateusz Bernacik
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type { NavigatorOptions } from './NavigatorOptions';
+import type { SeriesOptions } from '../../Core/Series/SeriesOptions';
+import Chart from '../../Core/Chart/Chart.js';
+import Navigator from './Navigator.js';
+import G from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+import Axis from '../../Core/Axis/Axis.js';
+import StandaloneNavigatorDefaults from './StandaloneNavigatorDefaults.js';
+const {
+    merge,
+    addEvent,
+    fireEvent,
+    pick
+} = U;
+
+declare module '../../Core/GlobalsLike.d.ts' {
+    interface GlobalsLike {
+        navigators: Array<StandaloneNavigator>;
+    }
+}
+
+type StandaloneNavigatorOptions = {
+    navigator: NavigatorOptions;
```

Comment:
I'm not sure if we can reuse the whole interface. Some options (`baseSeries`, `enabled` and `opposite`) shouldn't be available in Standalone. How about split into two? Like this:

```ts
interface BaseNavO {
  ...
};

// Current, in chart options:
interface NavigatorOptions extends BaseNavO {
  enabled: ...
  baseSeries: ...
};

// New interface for standalone nav:
interface StandaloneNavigatorOptions extends BaseNavO {
  ...
};
```

Reply:
In general, the structure of options seems to be a bit off: `Highcharts.navigator('container', options)` -> options should be options for the navigator, not the whole ChartOptions